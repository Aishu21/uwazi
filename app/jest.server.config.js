module.exports = {
  name: 'server',
  displayName: 'Server',
  testMatch: ['**/api/**/specs/*spec.(j|t)s?(x)', '**/shared/**/specs/*spec.(j|t)s?(x)'],
  testEnvironment: 'node',
  setupFilesAfterEnv: ['<rootDir>/setUpJestServer.js'],
  collectCoverageFrom: ['app/**/*.{js,ts}'],
  coverageDirectory: 'coverage',
  coverageReporters: ['none'],
};
