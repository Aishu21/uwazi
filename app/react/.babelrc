{
  "presets": ["es2015", "react"],
  "plugins": [
    ["babel-plugin-module-alias", [
      { "src": "./app/react", "expose": "~" },
      { "src": "./app/shared", "expose": "shared" },
    ]],
    "transform-class-properties",
    "add-module-exports"
  ]
}
