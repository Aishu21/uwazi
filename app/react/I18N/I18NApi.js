import api from 'app/utils/api';

export default {
  get() {
    return api.get('i18N/translations')
    .then((response) => {
      return response.json.rows;
    });
  },

  save(translation) {
    return api.post('i18N/translations', translation)
    .then((response) => {
      return response.json;
    });
  }
};
