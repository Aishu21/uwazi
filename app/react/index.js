import React from 'react';
import ReactDOM from 'react-dom';
import {browserHistory} from 'react-router';
import {Router} from 'react-router';
import Routes from './Routes';
import {Provider} from 'react-redux';
import CustomProvider from './App/Provider';
import store from './store';
import {AppContainer} from 'react-hot-loader';

const App = () => {
  return (
    <Provider store={store()}>
      <CustomProvider>
        <Router history={browserHistory}>{Routes}</Router>
      </CustomProvider>
    </Provider>
  );
};

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <App/>
    </AppContainer>,
    document.getElementById('root')
  );
};

render();
if (module.hot) {
  module.hot.accept();
}

import './sockets';

