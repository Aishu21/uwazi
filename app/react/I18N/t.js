import {store} from 'app/store';

let t = (context, key, _text) => {
  let text = _text || key;
  let state = store.getState();
  let translations = state.translations.toJS();
  let translation = translations.find((d) => d.locale === state.locale) || {values: {}};
  let translationExists = translation.values[context] && translation.values[context][key];

  return translationExists ? translation.values[context][key] : text;
};

export default t;
