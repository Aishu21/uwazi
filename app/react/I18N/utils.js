import * as Cookie from 'tiny-cookie';
import { isClient } from 'app/utils';

const I18NUtils = {
  getURLLocale: (locale, languages = []) => languages.find(l => l.key === locale) ? locale : null,

  getCookieLocale: () => {
    if (isClient && Cookie.get('locale')) {
      return Cookie.get('locale');
    }

    return null;
    // return I18NUtils.getLocaleIfExists(cookie.locale, languages);
  },

  getDefaultLocale: languages => (languages.find(language => language.default) || {}).key,

  getLocale(urlLanguage, languages = [], cookie = {}) {
    const newLanguage = I18NUtils.getURLLocale(urlLanguage, languages) ||
                        I18NUtils.getCookieLocale(cookie) ||
                        I18NUtils.getDefaultLocale(languages);
    this.saveLocale(newLanguage);
    return newLanguage;
  },

  saveLocale: (locale) => {
    if (isClient) {
      Cookie.set('locale', locale, { expires: 365 * 10 });
    }
  }
};

export default I18NUtils;
