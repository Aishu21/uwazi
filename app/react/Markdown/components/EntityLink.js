import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { I18NLink } from 'app/I18N';
import PagesContext from './Context';


export default class EntityLink extends Component {
  renderLink(entity) {
    const { children } = this.props;
    const sharedId = entity.toJS ? entity.get('sharedId') : entity.sharedId;
    const file = entity.toJS ? entity.get('file') : entity.file;
    const isEntity = !file;
    const type = isEntity ? 'entity' : 'document';

    const url = `/${type}/${sharedId}`;
    return (
      <I18NLink to={url} onClick={this.onClick}>
        {children}
      </I18NLink>
    );
  }

  render() {
    return (
      <PagesContext.Consumer>
        {entity => this.renderLink(entity)}
      </PagesContext.Consumer>
    );
  }
}

EntityLink.defaultProps = {
  children: '',
};

EntityLink.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]),
};
