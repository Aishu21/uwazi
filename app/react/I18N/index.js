import I18NMenu from './components/I18NMenu';
import I18NLink from './components/I18NLink';
import I18N from './components/I18N';
import t from './components/t';
import I18NUtils from './utils';

export default {
  I18NMenu,
  I18NLink,
  I18NUtils,
  I18N,
  t
};
