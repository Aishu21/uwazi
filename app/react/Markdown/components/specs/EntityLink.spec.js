/** @format */

import React from 'react';
import { shallow } from 'enzyme';

import { Provider } from 'react-redux';
import { createStore } from 'redux';
import EntityLink from '../EntityLink.js';
import PagesContext from '../Context';
import { I18NLink } from 'app/I18N';

describe('EntityLink', () => {
  let props;
  let entity;
  const store = createStore(() => ({}));

  beforeEach(() => {
    entity = { _id: '123', sharedId: 'abc' };
    props = {
      children: 'I want this as the link content',
    };
  });

  it('should generate a link based on the entity sharedId and if it has file or not', () => {
    const component = shallow(
      <Provider store={store}>
        <PagesContext.Provider value={entity}>
          <EntityLink {...props} />
        </PagesContext.Provider>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
