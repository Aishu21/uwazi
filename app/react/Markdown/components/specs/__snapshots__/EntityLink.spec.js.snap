// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityLink should generate a link based on the entity sharedId and if it has file or not 1`] = `
<[object Object]
  value={
    Object {
      "_id": "123",
      "sharedId": "abc",
    }
  }
>
  <EntityLink>
    I want this as the link content
  </EntityLink>
</[object Object]>
`;
